pkg/lang/chunkbuffer.go

```diff
@@ -27,9 +27,13 @@ func (cb *ChunkBuffer) Add(item interface{}) {
 }
 
 func (cb *ChunkBuffer) Chunk() IChunk {
-	ret := NewSliceChunk(cb.buffer)
+	newSlice := make([]interface{}, cb.end)
+	copy(newSlice, cb.buffer[:cb.end])
```

Comment:
following [clojure's implementation](https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/ChunkBuffer.java#L29), i recommend we reuse the underlying buffer, but with the correct view: `NewSliceChunk(cb.buffer[:cb.end])`. we remove the the `ChunkBuffer`'s reference to it below here so there's no need to allocate a new slice and copy.

Reply:
True! 😅 
I've made a `fixup!` commit to address this, thanks